ttnn/ttnn/operations/ternary.py

```diff
@@ -21,99 +21,58 @@ def torch_mac(input, tensor1, tensor2):
 
-            Returns tensor with the {name} of all of elements of the input tensors input, tensor1, tensor2.
 
-            .. math::
-                {name.replace('_',' ')}(\\mathrm{{input\\_tensor}}_i)
+def _golden_function_addcdiv(input_tensor_a, input_tensor_b, input_tensor_c, *args, value=1, **kwargs):
+    import torch
 
-            Args:
-                * :attr:`input_tensor`
-                * :attr:`input_tensor1`
-                * :attr:`input_tensor2`
+    return torch.addcdiv(input_tensor_a, input_tensor_b, input_tensor_c, value=value)
 
-            Example::
 
-                >>> tensor = ttnn.from_torch(torch.tensor((1, 2), dtype=torch.bfloat16), device=device)
-                >>> tensor1 = ttnn.from_torch(torch.tensor((1, 2), dtype=torch.bfloat16), device=device)
-                >>> tensor2 = ttnn.from_torch(torch.tensor((1, 2), dtype=torch.bfloat16), device=device)
-                >>> output = ttnn.{name}(tensor, tensor1, tensor2)
+ttnn.attach_golden_function(ttnn.addcdiv, golden_function=_golden_function_addcdiv)
 
-            """
 
-    @ttnn.register_python_operation(
-        name=f"ttnn.{name}",
-        golden_function=_golden_function,
-        doc=doc,
-    )
-    def ternary_function(
-        input: ttnn.Tensor,
-        input1: Union[ttnn.Tensor, float],
-        input2: Union[ttnn.Tensor, float],
-        *,
-        memory_config: ttnn.MemoryConfig = ttnn.DRAM_MEMORY_CONFIG,
-    ) -> ttnn.Tensor:
-        original_shape = input.shape
-        input_tensor = ttnn.unsqueeze_to_4D(input)
-        if not _is_scalar(input1) and not _is_scalar(input2):
-            input_tensor1 = ttnn.unsqueeze_to_4D(input1)
-            input_tensor2 = ttnn.unsqueeze_to_4D(input2)
-        elif _is_scalar(input1) and _is_scalar(input2):
-            input_tensor1 = input1
-            input_tensor2 = input2
+def _golden_function_lerp(input_tensor_a, input_tensor_b, input_tensor_c, *args, **kwargs):
+    import torch
 
-        if not isinstance(input_tensor, ttnn.Tensor):
-            raise TypeError("Expected input argument to be a ttnn.Tensor")
+    return torch.lerp(input_tensor_a, input_tensor_b, input_tensor_c)
 
-        if (isinstance(input_tensor1, ttnn.Tensor) and not isinstance(input_tensor2, ttnn.Tensor)) or (
-            not isinstance(input_tensor1, ttnn.Tensor) and isinstance(input_tensor2, ttnn.Tensor)
-        ):
-            raise TypeError("Expected other two inputs as either both tensor or scalar")
 
-        if not ttnn.is_tensor_storage_on_device(input_tensor):
-            raise RuntimeError("input_tensor must be on device!")
+ttnn.attach_golden_function(ttnn.lerp, golden_function=_golden_function_lerp)
 
-        output_tensor = ttl_ternary_function(input_tensor, input_tensor1, input_tensor2)
-        output_tensor = ttnn.reshape(output_tensor, original_shape)
-        return output_tensor
 
+def _golden_function_lerp_overload(input_tensor_a, input_tensor_b, input_tensor_c, *arg, **kwargs):
```

Comment:
no need for overload in golden fns. it uses no datatype in particular. only one golden fn for one api i.e. ttnn.lerp